### ISSUE TITLE
Spyder cannot load matplotlib if the latter tries to use PyQt5

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#2061]
_From [scallega...@gmail.com](https://code.google.com/u/100166133241400627403/) on 2014-11-21T12:03:40Z_

If you install (or upgrade) from source matplotlib on a system where PyQt5 is available, then matplotlib will happily configure itself to use PyQt5 and make it a preferred default over PyQt4.

Unfortunately, when PyQt5 is chosen as the backend, matplotlib cannot be used in spyder. 

What steps will reproduce the problem?

 1. Configure matplotlib to use the qt5agg backend
2. Start spyder
3. In the spyder ipython console type `from matplotlib import pyplot`

What is the expected output?

The matplotlib pyplot module should be imported and initialized

What do you see instead?

Traceback (most recent call last):

```
  File "<ipython-input-1-a371ea114746>", line 1, in \<module>
    from matplotlib import pyplot

  File "/home/callegar/.local/lib/python2.7/site-packages/matplotlib/pyplot.py", line 109, in \<module>
    _backend_mod, new_figure_manager, draw_if_interactive, _show = pylab_setup()

  File "/home/callegar/.local/lib/python2.7/site-packages/matplotlib/backends/__init__.py", line 32, in pylab_setup
    globals(),locals(),[backend_name],0)

  File "/home/callegar/.local/lib/python2.7/site-packages/matplotlib/backends/backend_qt5agg.py", line 18, in \<module>
    from .backend_qt5 import QtCore

  File "/home/callegar/.local/lib/python2.7/site-packages/matplotlib/backends/backend_qt5.py", line 27, in \<module>
    import matplotlib.backends.qt_editor.figureoptions as figureoptions

  File "/home/callegar/.local/lib/python2.7/site-packages/matplotlib/backends/qt_editor/figureoptions.py", line 17, in \<module>
    import matplotlib.backends.qt_editor.formlayout as formlayout

  File "/home/callegar/.local/lib/python2.7/site-packages/matplotlib/backends/qt_editor/formlayout.py", line 58, in \<module>
    from matplotlib.backends.qt_compat import QtGui, QtWidgets, QtCore

  File "/home/callegar/.local/lib/python2.7/site-packages/matplotlib/backends/qt_compat.py", line 111, in \<module>
    from PyQt5 import QtCore, QtGui, QtWidgets

RuntimeError: the PyQt5.QtCore module failed to register with the sip module
```

What version of the product are you using? On what operating system? Happening with spyder 2.3.1 on a kubuntu utopic 64bit machine, with matplotlib upgraded to 1.4.2 via pip.

Note that from an ipython console out of spyder everything is fine.

As a workaround, a qt4agg default backend can be made the default in .config/matplotlib/matplotlibrc. However, this is not the optimal solution. If spyder can only work with the qt4agg backend, it would better provide its own matplotlib defaults not force one to use them systemwide. 

Please provide any additional information below

.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=2061_

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2014-11-21T10:50:18Z_

Thanks for reporting. This seems really bad and it seems to be another blocker for 2.3.2.

**Labels:** -Priority-Medium Priority-Critical MS-v2.3.2 Cat-IPythonConsole

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2014-11-21T13:05:38Z_

After a bit of research, this seems a problem in Matplotlib itself. Let's wait until 2.3.3 to see if the mpl guys fix this problem in their next release.

**Labels:** -MS-v2.3.2 MS-v2.3.3

**Comment by spyder-issues-migrator** [comment]
_From [scallega...@gmail.com](https://code.google.com/u/100166133241400627403/) on 2014-11-23T04:14:27Z_

Is there a bug open in matplotlib to follow?

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2014-11-24T17:48:22Z_

Nop but I read the same report in the Anaconda mailing list (but not related to Spyder). Could you open an issue in the matplotlib tracker and reference it here?

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2015-01-08T14:58:47Z_

**Labels:** -MS-v2.3.3 MS-v2.4

**Comment by ccordoba12** [comment]
This was solved in PR #2226.
